src/app/theme/SplashScreenService.ts

```diff
@@ -72,28 +78,155 @@ export const showSplashScreen = (
+  }
 }
 
-/**
- * Retrieves the status of the splash screen ("visible" | "hidden" | "transitioning").
- * @returns A promise that resolves to the visibility status of the splash screen.
- */
-export const getSplashScreenStatus = (): Promise<VisibilityStatus> => {
-  return RNBootSplash.getVisibilityStatus()
+export const setTimeoutForSplashScreen = (
+  bootsplashName: SplashScreenEnum | undefined = splashScreens.GLOBAL
+): void => {
+  if (bootsplashName === splashScreens.SECURE_BACKGROUND) {
+    splashScreenLogger.info(
+      `Skipping timeout for secure background "${bootsplashName}"`
+    )
+    return
+  }
+
+  destroyTimer(bootsplashName)
+
+  autoHideTimers[bootsplashName] = setTimeout(() => {
+    splashScreenLogger.warn(
+      `Auto-hide duration reached for splash screen "${bootsplashName}"`
+    )
+
+    logToSentry(
+      new Error(
+        `Splashscreen reached autoHideDuration with bootsplashName "${bootsplashName}"`
+      )
+    )
+
+    void manageTimersAndHideSplashScreen(bootsplashName, true)
+  }, autoHideDuration)
+
+  splashScreenLogger.debug(
+    `Setting timeout with ID "${JSON.stringify(
+      autoHideTimers[bootsplashName]
+    )}" for splash screen "${bootsplashName}"`
+  )
+}
+
+const manageTimersAndHideSplashScreen = async (
+  bootsplashName: SplashScreenEnum | undefined = splashScreens.GLOBAL,
+  fromTimeout = false
+): Promise<void> => {
+  if (bootsplashName !== splashScreens.SECURE_BACKGROUND)
+    destroyTimer(bootsplashName, fromTimeout)
+
+  try {
+    await RNBootSplash.hide({ fade: true, bootsplashName })
+  } catch (error) {
+    splashScreenLogger.error(
+      `Error managing timers and hiding splash screen "${bootsplashName}"`,
+      error
+    )
+    logToSentry(error)
+  }
+}
+
+const destroyTimer = (
+  bootsplashName: SplashScreenEnum,
+  fromTimeout = false
+): void => {
+  const timer = autoHideTimers[bootsplashName]
+
+  if (autoHideTimers[bootsplashName]) {
+    if (fromTimeout) {
+      splashScreenLogger.debug(
+        `Destroying existing timer with ID "${JSON.stringify(
+          timer
+        )}" for splash screen "${bootsplashName}" after auto-hide duration reached`
+      )
+    } else {
+      splashScreenLogger.debug(
+        `Clearing existing timer with ID "${JSON.stringify(
+          timer
+        )}" for splash screen "${bootsplashName}" after manual hide`
+      )
+    }
+
+    clearTimeout(autoHideTimers[bootsplashName])
+    Reflect.deleteProperty(autoHideTimers, bootsplashName)
+  }
 }
+
+let activeTimersAtBackground: Record<string, boolean> = {}
+
+const resetTimersOnActive = (): void => {
+  splashScreenLogger.debug(
+    `App is becoming active. Evaluating timers to reset...`
+  )
+
+  const timersToReset = Object.keys(activeTimersAtBackground)
+
+  if (timersToReset.length === 0) {
+    splashScreenLogger.info(`No splash screen timers to reset.`)
+  } else {
+    timersToReset.forEach(splashScreen => {
+      if (activeTimersAtBackground[splashScreen]) {
```

Comment:
Why do we need this check? Seems like no code can set this value to false, and because `timersToReset` is binded to `activeTimersAtBackground` keys, seems like this condition is always `true`?